src/components/header/Header.tsx

```diff
@@ -1,16 +1,42 @@
 import styles from './Header.css';
 
-import React, { FC } from 'react';
+import React, { FC, useState, useEffect } from 'react';
 import { NavLink } from 'react-router-dom';
+import { getCity } from 'services/getCity';
+import HTTPRequest from 'services/httpService';
+import { LocationInfoType } from 'types/cityInfoType';
 
 type LinkType = {
   isActive: boolean;
 };
 
+let timeoutId: number;
+
 const setActive = ({ isActive }: LinkType) =>
   isActive ? `${styles.listItem} ${styles['active-link']}` : styles.listItem;
 
 const Header: FC = () => {
+  const [searchText, setSearchText] = useState<string>('');
+  const [cities, setCities] = useState<LocationInfoType[]>([]);
+
+  const getCities = async (search: string): Promise<{ locations: LocationInfoType[] }> => {
+    const result = await HTTPRequest(`/api/v1/location/search/${search}`, {});
+    return result;
+  };
+
+  useEffect(() => {
+    clearTimeout(timeoutId);
+
+    if (searchText) {
+      timeoutId = window.setTimeout(
+        () => getCities(searchText).then(res => setCities(res.locations)),
+        1000
+      );
```

Comment:
timeoutId defined out of component body looks not very good. this code can be written a bit more elegant way using react hooks, please try to refactor